mmv1/products/workstations/WorkstationConfig.yaml

```diff
@@ -196,96 +201,111 @@ properties:
           - !ruby/object:Api::Type::Array
             name: "tags"
             item_type: Api::Type::String
-            immutable: true
             description: |
               Network tags to add to the Compute Engine machines backing the Workstations.
           - !ruby/object:Api::Type::Boolean
             name: "disablePublicIpAddresses"
-            immutable: true
             description: |
               Whether instances have no public IP address.
           - !ruby/object:Api::Type::NestedObject
             name: "shieldedInstanceConfig"
             description: |
               A set of Compute Engine Shielded instance options.
-            immutable: true
+            diff_suppress_func: 'EmptyOrUnsetBlockDiffSuppress'
```

Comment:
Based on the note of `EmptyOrUnsetBlockDiffSuppress`, we strongly recommend against using it, as it may behave weirdly in some scenarios and we've seen cases that it's causing issues. However, it does make sense that this DSF will work in this case.

We could use custom flatteners for this field instead, which should add the false object explicitly to TF if the API does not return it.
It'd be something similar to https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/terraform/custom_flatten/health_check_log_config.go.erb. Plus, we may need to add `default_from_api: true` to the object if we decide to use  custom flatteners 

Adding @rileykarson to weigh in on this - if we'd prefer to move forward with the DSF or change to custom flatteners.

context: This is to solve that the API does not return the object even when
```
shielded_instance_config {
	enable_secure_boot          = false
	enable_vtpm                 = false
	enable_integrity_monitoring = false
}
```
is set explicitly.

Reply:
I've added a custom flattener and tested it. Copying the below from discussing with Riley, but I am re-testing with `default_from_api`, and it looks like we get different behavior when unsetting properties within the block.

Assuming a scenario in which the remote resource is in the following state:
```
shielded_instance_config {
    enable_secure_boot          = true
    enable_vtpm                 = true
    enable_integrity_monitoring = true
}
```

When using `default_from_api`:
  - emptying the block: `shielded_instance_config {}` does nothing (unexpected, all of the `enables_*` continue to be true in the background)
  - deleting the entire `shielded_instance_config` block sets all of the fields to null, causing API to set them to false (good)

Without `default_from_api`:
  - emptying the block: `shielded_instance_config {}` sets all the fields to null (good)
  - deleting the entire `shielded_instance_config` block sets all of the fields to null (also good)

I think the second example makes more sense with these fields, since they should always be false when they aren't present. Should I still be adding `default_from_api`?

Reply:
Are you setting default_from_api on each individual field within the `shielded_instance_config` block? I would think that setting default_from_api on the `shielded_instance_config` itself would make the behavior when an empty block or no block is specified be: resource does not send the `shielded_instance_config` value at all, preserving whatever the API has currently.

That may not be desired behavior, but it would be consistent.

Reply:
Thanks for taking a look! I had been testing with individual fields, but that explanation makes sense. I pushed a change to make the whole block to be `default_from_api`, and will do some manual tests while waiting for the other tests to run.

Reply:
It seems to work well -- closing this unless there are additional concerns.